applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_conn.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_mgmt_scan_for_conn_internal.h"
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+
+#include "bt_mgmt.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(bt_mgmt_scan);
+
+#define CONNECTION_PARAMETERS                                                                      \
+	BT_LE_CONN_PARAM(CONFIG_BLE_ACL_CONN_INTERVAL, CONFIG_BLE_ACL_CONN_INTERVAL,               \
+			 CONFIG_BLE_ACL_SLAVE_LATENCY, CONFIG_BLE_ACL_SUP_TIMEOUT)
+
+static uint8_t bonded_num;
+struct bt_le_scan_cb scan_callback;
+static bool cb_registered;
+static char const *srch_name;
+
+static void bond_check(const struct bt_bond_info *info, void *user_data)
+{
+	char addr_buf[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(&info->addr, addr_buf, BT_ADDR_LE_STR_LEN);
+
+	LOG_DBG("Stored bonding found: %s", addr_buf);
```

Comment:
It's not because of the code in this function but because the function is even called. It is called here: ``bt_foreach_bond(BT_ID_DEFAULT, bond_check, NULL);``, so that means that if `bond_check()` is entered we have found a bond. The code in `bond_check()` only converts the address to a string, prints it, and then increments the `bonded_num`